dvc/repo/diff.py

```diff
@@ -148,11 +148,14 @@ def _dir_output_paths(repo_tree, output, targets=None):
 def _filter_missing(repo, paths):
     repo_tree = RepoTree(repo, stream=True)
     for path in paths:
-        metadata = repo_tree.metadata(path)
-        if metadata.is_dvc:
-            out = metadata.outs[0]
-            if out.status().get(str(out)) == "not in cache":
-                yield path
+        try:
+            metadata = repo_tree.metadata(path)
+            if metadata.is_dvc:
+                out = metadata.outs[0]
+                if out.status().get(str(out)) == "not in cache":
+                    yield path
+        except FileNotFoundError:
```

Comment:
This check is for filtering missing DVC outs from our set of potentially deleted paths, if a file isn't present in the workspace at all (no dvcfile or data) it must be deleted